webapi/Controllers/DocumentController.cs

```diff
@@ -283,122 +376,122 @@ private async Task ValidateDocumentImportFormAsync(DocumentImportForm documentIm
+            //            $"{OcrSupportOptions.PropertyName}:{nameof(OcrSupportOptions.Type)} is set to " +
+            //            nameof(OcrSupportOptions.OcrSupportType.None));
+            //    }
+            //    default:
+            //        throw new ArgumentException($"Unsupported file type: {fileType}");
+            //}
+
+            // $$$ ISIMAGE ???
+            if (documentImportForm.UseContentSafety)
             {
-                case SupportedFileType.Txt:
-                case SupportedFileType.Pdf:
-                case SupportedFileType.Md:
-                    break;
-                case SupportedFileType.Jpg:
-                case SupportedFileType.Png:
-                case SupportedFileType.Tiff:
-                    if (this._ocrSupportOptions.Type != OcrSupportOptions.OcrSupportType.None)
-                    {
-                        if (documentImportForm.UseContentSafety)
-                        {
-                            if (this._contentSafetyService == null || !this._contentSafetyService.ContentSafetyStatus(this._logger))
-                            {
-                                throw new ArgumentException("Unable to analyze image. Content Safety is currently disabled in the backend.");
-                            }
+                if (this._contentSafetyService == null || !this._contentSafetyService.ContentSafetyStatus(this._logger))
+                {
+                    throw new ArgumentException("Unable to analyze image. Content Safety is currently disabled in the backend.");
+                }
 
-                            var violations = new List<string>();
-                            try
-                            {
-                                // Call the content safety controller to analyze the image
-                                var imageAnalysisResponse = await this._contentSafetyService.ImageAnalysisAsync(formFile, default);
-                                violations = this._contentSafetyService.ParseViolatedCategories(imageAnalysisResponse, this._contentSafetyService.Options.ViolationThreshold);
-                            }
-                            catch (Exception ex) when (!ex.IsCriticalException())
-                            {
-                                this._logger.LogError(ex, "Failed to analyze image {0} with Content Safety. ErrorCode: {{1}}", formFile.FileName, (ex as AIException)?.ErrorCode);
-                                throw new AggregateException($"Failed to analyze image {formFile.FileName} with Content Safety.", ex);
-                            }
+                var violations = new List<string>();
+                try
+                {
+                    // Call the content safety controller to analyze the image
+                    var imageAnalysisResponse = await this._contentSafetyService.ImageAnalysisAsync(formFile, default);
```

Comment:
Does this work right now without the image type check? Or does it just throw an exception on any non-image file?

Reply:
Yeah...I think it'll throw.  Yes...its a hacky merge as I tried to expose a contract we could use via semantic-memory:

https://github.com/microsoft/semantic-memory/pull/29